src/utils/helpers.test.js

```diff
@@ -56,3 +69,123 @@ describe('getUKFormattedEuros', () => {
+    const actual = getCurrentMonth();
+
+    expect(actual).toBe(expected);
+  });
+  test('if the current date is 2024-02-29, it should return "February 2024"', () => {
+    const date = '2024-02-29';
+    global.Date.now = getDateNowStub(date);
+
+    const expected = 'February 2024';
+    const actual = getCurrentMonth();
+
+    expect(actual).toBe(expected);
+  });
+  test('if the current date is 2050-12-31, it should return "December 2050"', () => {
+    const date = '2050-12-31';
+    global.Date.now = getDateNowStub(date);
+
+    const expected = 'December 2050';
+    const actual = getCurrentMonth();
+
+    expect(actual).toBe(expected);
+  });
+});
+
+describe('getUniqueMonths', () => {
+  afterAll(() => {
+    global.Date.now = realDateNow;
+  });
+
+  it('should throw an error if no argument is passed', () => {
+    function passZeroArgs() {
+      getUniqueMonths();
+    }
+    expect(passZeroArgs).toThrowError('undefined is not an array');
+  });
+  it('should throw an error if passed a value that is not an array', () => {
+    function passString() {
+      getUniqueMonths('any string');
+    }
+    function passObject() {
+      getUniqueMonths({});
+    }
+    expect(passString).toThrowError('any string is not an array');
+    expect(passObject).toThrowError('[object Object] is not an array');
+  });
+  it('should throw an error if passed more than one argument', () => {
+    function passMoreThanOneArg() {
+      getUniqueMonths([], []);
+    }
+    expect(passMoreThanOneArg).toThrowError(
+      'Only one argument may be passed to the function.'
+    );
+  });
+  it('should return a new array', () => {
+    const inputArray = [];
+    const outputArray = getUniqueMonths(inputArray);
+    expect(outputArray).not.toBe(inputArray);
+  });
+  it('should return an array of strings in "Month YYYY" format', () => {
+    global.Date.now = getDateNowStub('2022-04-01');
+    const expenses = [
+      { title: 'Expense 1', date: '2022-04-01', amount: 100, id: 1 },
+      { title: 'Expense 2', date: '2022-06-01', amount: 100, id: 2 },
+    ];
+
+    const expected = ['April 2022', 'June 2022'];
+    const actual = getUniqueMonths(expenses);
+    expect(actual).toStrictEqual(expected);
+  });
+  it('should always return the current month', () => {
+    global.Date.now = getDateNowStub('2022-04-01');
+    const expenses = [];
+
+    const expected = ['April 2022'];
+    const actual = getUniqueMonths(expenses);
+
+    expect(actual).toStrictEqual(expected);
+  });
+  it('should not return duplicate months', () => {
+    global.Date.now = getDateNowStub('2022-04-01');
+    const expenses = [
+      { title: 'April 2022 expense 1', date: '2022-04-01', amount: 100, id: 1 },
+      { title: 'April 2022 expense 2', date: '2022-04-02', amount: 100, id: 2 },
+      { title: 'April 2021 expense', date: '2021-04-02', amount: 100, id: 3 },
+    ];
+
+    const expected = ['April 2021', 'April 2022'];
+    const actual = getUniqueMonths(expenses);
```

Comment:
something I realize only now: i think the `getUniqueMonths` is either named to unspecific (I would never expect to put a list of expenses into this function if I read the name) or it receives too specific parameters (imagine how we could re-use this function in other contexts if it would only take a list of dates).

changing to the latter, would also affect the tests and make them much easier.

what do you think?

Reply:
Sure, I agree it isn't very clear what this function does. Maybe we could rename the helper for now (e.g. `getUniqueExpenseMonths` or `getUniqueMonthsFromExpenses`) to add some clarity and then create a separate ticket to modify the parameters and update the app and test code accordingly? If you agree, I can go ahead and create the ticket and leave it open to whoever wants to tackle it.

Reply:
sure thing. let's do it this way.